packages/protocol-utils/src/orders.ts

```diff
@@ -129,6 +135,35 @@ export abstract class OrderBase {
         }
     }
 
+    /*
+     * This is an advanced order signature function that should only be used if you know what you're doing.
+     * It allows you to bring your own wallet signature callback function, this can be used to sign orders
+     * with a wallet that is otherwise unsupported by Web3Wrapper.
+     * The signature function needs to implement correct signing of the data and return the results as a string.
+     */
+    public async _getSignatureWithSignFunctionAsync(
+        signFn: SignatureCallbackFunction,
+        type: SignatureType,
+        signer: string = this.maker,
+    ): Promise<Signature> {
+        switch (type) {
+            case SignatureType.EIP712:
+                return eip712SignTypedDataWithSignatureFnAsync(
+                    this.getEIP712TypedData(),
+                    signer,
+                    (signFn as unknown) as EIP712SignCallbackFunction,
```

Comment:
This is pretty horrible but I couldn't figure out a good way to handle this with proper typing. Open to any suggestions for improvements 😅

Reply:
wow it really won't let you just do a direct cast?

Reply:
nah TS was completely freaking out over the lack of overlap 🙈